### PR TITLE
fix: self hosted register

### PR DESCRIPTION
**Pull request by darkskygit**

### CODE REVIEW

**Comment by darkskygit** [comment]
* **#6581** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6581?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @darkskygit and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/server/src/data/migrations/1713258139569-refresh-new-plan.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+export class RefreshNewPlan1713258139569 {
+  // do the migration
+  static async up(db: PrismaClient, ref: ModuleRef) {
+    const config = ref.get(Config, { strict: false });
+    const { AFFINE_ADMIN_EMAIL } = process.env;
+    if (config.isSelfhosted && AFFINE_ADMIN_EMAIL) {
+      const user = ref.get(UserService, { strict: false });
+      const quota = ref.get(QuotaService, { strict: false });
+      const { id } = (await user.findUserByEmail(AFFINE_ADMIN_EMAIL)) || {};
```
Comment (EYHN) [inline packages/backend/server/src/data/migrations/1713258139569-refresh-new-plan.ts:17]: Some selfhosted server have more than one user, we should let all users use this plan

File: packages/backend/server/src/data/migrations/99999999-self-host-admin.ts [hunk 2]
```diff
@@ -19,13 +21,17 @@ export class SelfHostAdmin99999999 {
...
+        process.env.AFFINE_ADMIN_EMAIL,
+        {
+          name: 'AFFINE First User',
+          emailVerifiedAt: new Date(),
+          password: await crypto.encryptPassword(
+            process.env.AFFINE_ADMIN_PASSWORD
+          ),
+        }
+      );
+      await quota.switchUserQuota(id, QuotaType.UnlimitedPlanV1);
```
Comment (EYHN) [inline packages/backend/server/src/data/migrations/99999999-self-host-admin.ts]: Can we add config item for default quota plan?

**Comment by Felitendo** [comment]
@forehalo

**Comment by darkskygit** [comment]
outdated, will impl this later

**Comment by Felitendo** [comment]
> outdated, will impl this later

What do you mean by "later". This has been an issue for 2 years

**Comment by darkskygit** [comment]
> > outdated, will impl this later
> 
> What do you mean by "later". This has been an issue for 2 years

the relevant module is currently being refactored, so the code in this pr will become invalid
a new pr will be submitted after the refactoring is completed

**Pull request closed without merging**